Java oop advansed solution

fed4ik opened the pull request.

fed4ik committed:
Created interface of figures with method toDraw()

fed4ik committed:
Created classes for creating random figure with random color and parameters

fed4ik committed:
Created figures that implement interface Figure

fed4ik committed:
created the array of figures and output information of the figure's list

aaananas reviewed (changes requested):
![image](https://user-images.githubusercontent.com/81781479/184478157-d99ad9ea-b64c-42a6-884a-7138b9f60a80.png)

fed4ik committed:
fixed class IsoscelesTrapezoid ( renamed class )

fed4ik commented:
T E S T S
-------------------------------------------------------

Results :

Tests run: 0, Failures: 0, Errors: 0, Skipped: 0

[INFO]
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ jv-oop-advanced ---
[INFO] Building jar: C:\Users\fed4i\IdeaProjects\jv-oop-advanced\target\jv-oop-advanced-1.0-SNAPSHOT.jar
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  8.083 s
[INFO] Finished at: 2022-08-13T12:58:37+03:00
[INFO] ------------------------------------------------------------------------

Анастасія привіт! Чомусь при перевірці "mvn clean package" маю результат без помилок))) Тревіс трохи матюкливий сьогодні, спробую пофіксити код!)

fed4ik committed:
replace interface Figure to abstract class Figure with a variable color

fed4ik committed:
added interface with method toDraw()

fed4ik committed:
fixed class

aaananas commented on src/main/java/core/basesyntax/Circle.java:
```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+public class Circle extends Figure {
+
```
```suggestion
```

aaananas commented on src/main/java/core/basesyntax/Figure.java:
```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+public abstract class Figure implements Figures {
```
```suggestion
public abstract class Figure implements Drawable, AreaCalculator {
```

aaananas commented on src/main/java/core/basesyntax/FigureList.java:
```diff
@@ -0,0 +1,9 @@
+package core.basesyntax;
+
+public enum FigureList {
```
delete this file

aaananas commented on src/main/java/core/basesyntax/Figures.java:
```diff
@@ -0,0 +1,5 @@
+package core.basesyntax;
+
+public interface Figures {
+    public void toDraw();
```
```suggestion
    public void draw();
```

aaananas commented on src/main/java/core/basesyntax/Figures.java:
```diff
@@ -0,0 +1,5 @@
+package core.basesyntax;
+
+public interface Figures {
```
```suggestion
public interface Drawable {
```

aaananas commented on src/main/java/core/basesyntax/RightTriangle.java:
```diff
@@ -0,0 +1,32 @@
+    @Override
+    public String toString() {
+        return "Figure: rightTriangle, area: "
+                + Math.sqrt(3) * side * side / 4
```
```suggestion
                + getArea()
```
add interface AreaCalculator with method getArea
and Override this method in each of your figures

aaananas commented on src/main/java/core/basesyntax/Rectangle.java:
```diff
@@ -0,0 +1,46 @@
+
+    @Override
+    public void toDraw() {
+        System.out.println(toString());
```
don't use toString here. Move everything from toString here
Change each class accordingly

aaananas commented on src/main/java/core/basesyntax/Rectangle.java:
```diff
@@ -0,0 +1,46 @@
+package core.basesyntax;
+
+public class Rectangle extends Figure {
+
+    private int sideA;
+    private int sideB;
```
```suggestion
    private int firstSide;
    private int secondSide;
```

aaananas commented on src/main/java/core/basesyntax/Rectangle.java:
```diff
@@ -0,0 +1,46 @@
+    private int sideA;
+    private int sideB;
+
+    public Rectangle(int sideA, int sideB, String color) {
+        this.sideA = sideA;
+        this.sideB = sideB;
+        this.color = color;
```
```suggestion
    public Rectangle(int sideA, int sideB, String color) {
        super(color);
        this.sideA = sideA;
        this.sideB = sideB;
```

aaananas commented on src/main/java/core/basesyntax/Square.java:
```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+public class Square extends Figure {
+    private int sideA;
```
```suggestion
    private int side;
```

aaananas commented on src/main/java/core/basesyntax/Square.java:
```diff
@@ -0,0 +1,33 @@
+    private int sideA;
+
+    public Square(int sideA, String color) {
+        this.sideA = sideA;
+        this.color = color;
```
```suggestion
        super(color);
        this.side = side;
```

aaananas commented on src/main/java/core/basesyntax/HelloWorld.java:
```diff
@@ -4,5 +4,20 @@
+        Figure[] figures = new Figure[6];
+        FigureSupplier figureSupplier = new FigureSupplier();
 
+        for (int i = 0; i < figures.length / 2; i++) {
+            figures[i] = figureSupplier.getRandomFigure();
+        }
+
+        for (int i = figures.length / 2; i < figures.length; i++) {
+            figures[i] = figureSupplier.getDefaultFigure();
+        }
```
combine into one for

aaananas commented on src/main/java/core/basesyntax/IsoscelesTrapezoid.java:
```diff
@@ -0,0 +1,57 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+    private int sideA;
+    private int sideB;
+    private int height;
```
```suggestion
    private int firstBase;
    private int secondBase;
    private int height;
```

aaananas commented on src/main/java/core/basesyntax/IsoscelesTrapezoid.java:
```diff
@@ -0,0 +1,57 @@
+    private int height;
+
+    public IsoscelesTrapezoid(int sideA, int sideB, int height, String color) {
+        this.sideA = sideA;
```
super(color)

fed4ik committed:
Created new interface with method getArea()

fed4ik committed:
Renamed from Figures and renamed method toDraw() to draw()

fed4ik committed:
- fixed constructor,
- deleted toString(),
- rewrote method draw(),
- implemented method getArea()

fed4ik committed:
- renamed parameter "sideA" to "side"
- fixed constructor,
- deleted toString(),
- rewrote method draw(),
- implemented method getArea()

fed4ik committed:
- fixed constructor,
- deleted toString(),
- rewrote method draw(),
- implemented method getArea()

fed4ik committed:
- renamed parameters sideA to firsSide and sideB to secondSide,
- fixed constructor,
- deleted toString(),
- rewrote method draw(),
- implemented method getArea()

fed4ik committed:
- renamed parameters,
- fixed constructor,
- deleted toString(),
- rewrote method draw(),
- implemented method getArea()

fed4ik committed:
combined into one cycle for

fed4ik committed:
replace if to switch-case

fed4ik committed:
deleted FigureList

fed4ik committed:
- added interface AreaCalculator
- added constructor

aaananas commented on src/main/java/core/basesyntax/HelloWorld.java:
```diff
@@ -4,5 +4,27 @@
+        Figure[] figures = new Figure[6];
+        FigureSupplier figureSupplier = new FigureSupplier();
 
+        for (int i = 0; i < figures.length; i++) {
+
+            if (i < figures.length / 2) {
+                figures[i] = figureSupplier.getRandomFigure();
+            }
+
+            if (i >= figures.length / 2) {
+                figures[i] = figureSupplier.getDefaultFigure();
+            }
+        }
```
```suggestion
        for (int i = 0; i < figures.length; i++) {
            figures[i] = (i < figures.length / 2) ? figureSupplier.getRandomFigure() 
                               : figureSupplier.getDefaultFigure();
        }
```

aaananas commented on src/main/java/core/basesyntax/HelloWorld.java:
```diff
@@ -4,5 +4,27 @@
+                figures[i] = figureSupplier.getDefaultFigure();
+            }
+        }
+
+        for (int i = figures.length / 2; i < figures.length; i++) {
+
+        }
+
```
```suggestion
```

aaananas commented on src/main/java/core/basesyntax/FigureSupplier.java:
```diff
@@ -0,0 +1,51 @@
+    public Figure getRandomFigure() {
+        Figure figure;
+        switch (random.nextInt(numberOfFigures)) {
+            case (0):
+                figure = new Square((random.nextInt(maxValuesOfUnits - minValuesOfUnits) + 1),
+                        new ColorSupplier().getRandomColor());
+                break;
+            case (1):
+                figure = new Rectangle((random.nextInt(maxValuesOfUnits - minValuesOfUnits) + 1),
+                        (random.nextInt(maxValuesOfUnits - minValuesOfUnits) + 1),
+                        new ColorSupplier().getRandomColor());
+                break;
+            case (2):
+                figure = new RightTriangle(
+                        (random.nextInt(maxValuesOfUnits - minValuesOfUnits) + 1),
+                        new ColorSupplier().getRandomColor());
+                break;
+            case (3):
+                figure = new Circle((random.nextInt(maxValuesOfUnits - minValuesOfUnits) + 1),
+                        new ColorSupplier().getRandomColor());
+                break;
+            case (4):
+                figure = new IsoscelesTrapezoid(
+                        (random.nextInt(maxValuesOfUnits - minValuesOfUnits) + 1),
+                        (random.nextInt(maxValuesOfUnits - minValuesOfUnits) + 1),
+                        (random.nextInt(maxValuesOfUnits - minValuesOfUnits) + 1),
+                        new ColorSupplier().getRandomColor());
+                break;
```
```suggestion
            case 0:
                return getSquare();
            case 1:
                return getRectangle);
                break;
            case 2:
                return getRightTriangle();
            case 3:
                return getCircle();
            case 4:
                return getIsoscelesTrapezoid()
```

code would be easier to read if you create methods to get figures

aaananas commented on src/main/java/core/basesyntax/FigureSupplier.java:
```diff
@@ -0,0 +1,51 @@
+                        (random.nextInt(maxValuesOfUnits - minValuesOfUnits) + 1),
+                        new ColorSupplier().getRandomColor());
+                break;
+            default:
+                figure = new Circle(10, "WHITE");
+                break;
```
```suggestion
            default:
                return getDefaultFigure();
```

aaananas commented on src/main/java/core/basesyntax/FigureSupplier.java:
```diff
@@ -0,0 +1,51 @@
+import java.util.Random;
+
+public class FigureSupplier {
+
```
```suggestion
```
remove empty line

fed4ik committed:
- deleted useless cycle "for"
- rewrote "if" via ternary operator

fed4ik committed:
rewrote "switch-case" via methods

fed4ik commented:
Your comments was very correct))))) Thanks)))))

aaananas commented on src/main/java/core/basesyntax/HelloWorld.java:
```diff
@@ -4,5 +4,18 @@
+        FigureSupplier figureSupplier = new FigureSupplier();
 
+        for (int i = 0; i < figures.length; i++) {
+
```
```suggestion
```

Imihar replied:
Doylhcyo7odulc

aaananas commented on src/main/java/core/basesyntax/FigureSupplier.java:
```diff
@@ -0,0 +1,62 @@
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int numberOfFigures = 4; // start count from zero
+    public static final int maxValuesOfUnits = 10;
+    public static final int minValuesOfUnits = 1;
```
```suggestion
    private static final int NUMBER_OF_FIGURES = 4;
    private static final int UPPER_BOUND = 10;
```
you are not a developer if you count not from 0 :)
There is no need for min value

aaananas commented on src/main/java/core/basesyntax/FigureSupplier.java:
```diff
@@ -0,0 +1,62 @@
+
+    public Figure getRandomFigure() {
+        switch (random.nextInt(numberOfFigures)) {
+            case (0):
+                return getSquare();
+            case (1):
+                return getRectangle();
+            case (2):
+                return getRightTriangle();
+            case (3):
+                return getCircle();
+            case (4):
```
```suggestion
            case 0:
                return getSquare();
            case 1:
                return getRectangle();
            case 2:
                return getRightTriangle();
            case 3:
                return getCircle();
            case 4:
```

Imihar replied:
Ludiyxhl6ie68dullhxyio5e7oe79ew5858wi6e7ei6elydjlfup

fed4ik committed:
deleted empty line

fed4ik committed:
renamed parameters
deleted parentheses
about count, good joke, I got it))))))
